app/modules/wallet/wallet.controller.ts

```diff
@@ -0,0 +1,68 @@
+import * as restify from 'restify';
+import { InternalServerError, NotFoundError } from 'restify-errors';
+import IController from '../../interfaces/utils/IController';
+import { IRequest, IResponse } from '../../interfaces/utils/IServer';
+import Wallet from '../../models/wallet.model';
+import WalletAddress from '../../models/wallet_address.model';
+import WalletUser from '../../models/wallet_user.model';
+import { IWallet } from '../../interfaces/models';
+
+export default class WalletsController implements IController {
+  public async post(req: IRequest, res: IResponse, next: restify.Next) {
+    try {
+      const walletObj = req.body;
+      const wallet = <IWallet> await Wallet.create({
+        ...walletObj,
+        client_id: req.client_id
+      });
+      return res.send(wallet);
+    } catch (e) {
+      req.log.error(e);
+      throw new InternalServerError(e);
+    }
+    return next();
+  }
+
+  public async get(req: IRequest, res: IResponse, next: restify.Next) {
+    try {
+      const wallet = await Wallet.findOne({
+        where: { id: req.params.wallet_id },
+        include: [
+          {
+            model: WalletAddress,
+            as: 'addresses',
+            attributes: ['id'],
+            where: { wallet_id: req.params.wallet_id }
+          },
+          {
+            model: WalletUser,
+            as: 'users',
+            attributes: ['id'],
+            where: { wallet_id: req.params.wallet_id }
```

Comment:
Don't need to give it a where clause, as it'll fetch only those which relates to given wallet_id ;)